packages/web3wallet/src/client.ts

```diff
@@ -0,0 +1,183 @@
+import EventEmitter from "events";
+import { CLIENT_CONTEXT } from "./constants";
+import { Engine, PendingRequest } from "./controllers";
+import { IWeb3Wallet, Web3WalletTypes } from "./types";
+
+export class Web3Wallet extends IWeb3Wallet {
+  public name: IWeb3Wallet["name"];
+  public core: IWeb3Wallet["core"];
+  public logger: IWeb3Wallet["logger"];
+  public events: IWeb3Wallet["events"] = new EventEmitter();
+  public engine: IWeb3Wallet["engine"];
+  public pendingRequest: IWeb3Wallet["pendingRequest"];
+  public metadata: IWeb3Wallet["metadata"];
+
+  static async init(opts: Web3WalletTypes.Options) {
+    const client = new Web3Wallet(opts);
+    await client.initialize();
+
+    return client;
+  }
+
+  constructor(opts: Web3WalletTypes.Options) {
+    super(opts);
+    this.metadata = opts.metadata as any; // TODO: automatic fetch of metadata when?
+    this.name = opts.name || CLIENT_CONTEXT;
+    this.core = opts.core;
+    this.logger = this.core.logger;
+    this.engine = new Engine(this);
+    this.pendingRequest = new PendingRequest(this.core, this.logger);
+  }
+
+  // ---------- Events ----------------------------------------------- //
+
+  public on: IWeb3Wallet["on"] = (name, listener) => {
+    return this.events.on(name, listener);
+  };
+
+  public once: IWeb3Wallet["once"] = (name, listener) => {
+    return this.events.once(name, listener);
+  };
+
+  public off: IWeb3Wallet["off"] = (name, listener) => {
+    return this.events.off(name, listener);
+  };
+
+  public removeListener: IWeb3Wallet["removeListener"] = (name, listener) => {
+    return this.events.removeListener(name, listener);
+  };
+
+  // ---------- Engine ----------------------------------------------- //
+
+  public approveSession: IWeb3Wallet["approveSession"] = async (params) => {
+    try {
+      return await this.engine.approveSession(params);
+    } catch (error: any) {
```

Comment:
Given that we are catching and logging all errors in sub-sdks like sign and auth, do we need to re-do this here?

Reply:
technically the SDK doesn't know that/if errors are being caught at an inner level so I don't think there is harm be on the safe side